logstash-core/src/main/java/org/logstash/ackedqueue/Queue.java

```diff
@@ -290,7 +300,7 @@ public long write(Queueable element) throws IOException {
     public boolean isFull() {
         // TODO: I am not sure if having unreadCount as volatile is sufficient here. all unreadCount updates are done inside syncronized
         // TODO: sections, I believe that to only read the value here, having it as volatile is sufficient?
-        return (this.maxUnread > 0) ? this.unreadCount >= this.maxUnread : false;
+        return (((this.maxUnread > 0) ? this.unreadCount >= this.maxUnread : false) || (this.currentSize >= this.maxSize));
```

Comment:
mixing the ternary operator with a boolean `||` expression is hard to reason, I would refactor this using `if`/`else` ...